File: java/utility/ResultBody.java
package voxgig.solardemosdk.utility;

import voxgig.solardemosdk.core.Context;
import voxgig.solardemosdk.core.Response;
import voxgig.solardemosdk.core.Result;

final class ResultBody {

  private ResultBody() {}

  static Result resultBody(Context ctx) {
    Response response = ctx.response;
    Result result = ctx.result;

    if (result != null) {
      if (response != null && response.jsonFunc != null && response.body != null) {
        result.body = response.jsonFunc.get();
      }
    }

    return result;
  }
}

File: php/utility/ResultBody.php
<?php
declare(strict_types=1);

// Solardemo SDK utility: result_body

class SolardemoResultBody
{
    public static function call(SolardemoContext $ctx): ?SolardemoResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result && $response && $response->json_func && $response->body) {
            $result->body = ($response->json_func)();
        }
        return $result;
    }
}
